storagenode/piecestore/endpoint.go

```diff
@@ -166,6 +169,48 @@ func (endpoint *Endpoint) Delete(ctx context.Context, delete *pb.PieceDeleteRequ
 	return &pb.PieceDeleteResponse{}, nil
 }
 
+// DeletePiece handles deleting a piece on piece store requested by satellite.
+//
+// It doesn't return an error if the piece isn't found by any reason.
+func (endpoint *Endpoint) DeletePiece(
+	ctx context.Context, req *pb.PieceDeletePieceRequest,
+) (_ *pb.PieceDeletePieceResponse, err error) {
+	defer mon.Task()(&ctx, req.PieceId.String())(&err)
```

Comment:
I am not very well familiar with the monitoring, but noticed that the invokation here differs from the one in the old deprecated method. Is there a reason for this?

Reply:
In my opinion, we should add to monitoring the arguments of a function for having more insights of every call and a `Task` is designed for so (https://godoc.org/gopkg.in/spacemonkeygo/monkit.v2#Func.Task).

Obviously, I'm not sending the argument as it's, because I think that it's much easier to see the pice ID in its string format in the trace rather than the `protobuff` struct which only contains the piece ID.

Why aren't we passing the arguments of the function in other `monkit` Task calls?

Well, I don't know. I asked in slack a few months ago if passing the arguments will impact substantially the performance and the answer was "it shouldn't".

Since that answer, I've been commenting on reviews to pass arguments to `monkit Tasks` when I think that the parameters can provide insights to the traces.

Reply:
OK. Thanks for explaining! Should we update our style guide with this info? https://github.com/storj/docs/blob/master/code/Style.md#telemetry

Reply:
That's a good idea.
You made me think that I should have asked this in arch meeting, but it may be too late because this was several months ago and I started to comment in reviews.

I will update that documentation and if anybody is opposite, we'll be able to see in the review and if a discussion starts I will take care of binging up the topic in the next arch meeting.

Reply:
https://github.com/storj/docs/pull/128